app/assets/javascripts/app-level.js

```diff
@@ -0,0 +1,15 @@
+/* JavaScript code that will be added to every page throughout the application */
+
+/* Moment JS generic converter */
+$(".moment-time").each(function(ind) {
```

Comment:
One thing you could do is have a couple of helper classes, say `.moment-long-date`, `.moment-short-date`, `.moment-time`, and then one `.moment-custom-time` class. For the first three, you'd hard-code the appropriate format, and then for the `.moment-custom-time` class you'd search for the `data-format` attribute.

This makes making the change I suggested below (re-ordering the way the date information is presented) a little easier down the road (one change in one place vs. find/replace).